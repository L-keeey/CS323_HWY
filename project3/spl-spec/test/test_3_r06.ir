Error type 2 at Line 6: function should be invoked with a definition.
Error type 2 at Line 8: function should be invoked with a definition.
FUNCTION hanoi :
PARAM v1
PARAM v2
PARAM v3
PARAM v4
t1 := v1
t2 := #1
IF t1 == t2 GOTO label1
GOTO label2
LABEL label1 :
t6 := v2
t7 := #1000000
t4 := t6 * t7
t5 := v4
t3 := t4 + t5
WRITE t3
GOTO label3
LABEL label2 :
t10 := v1
t11 := #1
t9 := t10 - t11
t12 := v2
t13 := v4
t14 := v3
ARG t14
ARG t13
ARG t12
ARG t9
t8 := CALL hanoi
t18 := v2
t19 := #1000000
t16 := t18 * t19
t17 := v4
t15 := t16 + t17
WRITE t15
t22 := v1
t23 := #1
t21 := t22 - t23
t24 := v3
t25 := v2
t26 := v4
ARG t26
ARG t25
ARG t24
ARG t21
t20 := CALL hanoi
LABEL label3 :
t27 := #0
RETURN t27
FUNCTION main :
t28 := #3
v5 := t28
t30 := v5
t31 := #1
t32 := #2
t33 := #3
ARG t33
ARG t32
ARG t31
ARG t30
t29 := CALL hanoi
t34 := #0
RETURN t34
