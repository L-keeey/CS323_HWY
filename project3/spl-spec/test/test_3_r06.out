Error type 2 at Line 6: function should be invoked with a definition.
Error type 2 at Line 8: function should be invoked with a definition.
Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      FunDec (1)
        ID: hanoi
        LP
        VarList (1)
          ParamDec (1)
            Specifier (1)
              TYPE: int
            VarDec (1)
              ID: n
          COMMA
          VarList (1)
            ParamDec (1)
              Specifier (1)
                TYPE: int
              VarDec (1)
                ID: p1
            COMMA
            VarList (1)
              ParamDec (1)
                Specifier (1)
                  TYPE: int
                VarDec (1)
                  ID: p2
              COMMA
              VarList (1)
                ParamDec (1)
                  Specifier (1)
                    TYPE: int
                  VarDec (1)
                    ID: p3
        RP
      CompSt (1)
        LC
        StmtList (2)
          Stmt (2)
            IF
            LP
            Exp (2)
              Exp (2)
                ID: n
              EQ
              Exp (2)
                INT: 1
            RP
            Stmt (2)
              CompSt (2)
                LC
                StmtList (3)
                  Stmt (3)
                    Exp (3)
                      ID: write
                      LP
                      Args (3)
                        Exp (3)
                          Exp (3)
                            Exp (3)
                              ID: p1
                            MUL
                            Exp (3)
                              INT: 1000000
                          PLUS
                          Exp (3)
                            ID: p3
                      RP
                    SEMI
                RC
            ELSE
            Stmt (5)
              CompSt (5)
                LC
                StmtList (6)
                  Stmt (6)
                    Exp (6)
                      ID: hanoi
                      LP
                      Args (6)
                        Exp (6)
                          Exp (6)
                            ID: n
                          MINUS
                          Exp (6)
                            INT: 1
                        COMMA
                        Args (6)
                          Exp (6)
                            ID: p1
                          COMMA
                          Args (6)
                            Exp (6)
                              ID: p3
                            COMMA
                            Args (6)
                              Exp (6)
                                ID: p2
                      RP
                    SEMI
                  StmtList (7)
                    Stmt (7)
                      Exp (7)
                        ID: write
                        LP
                        Args (7)
                          Exp (7)
                            Exp (7)
                              Exp (7)
                                ID: p1
                              MUL
                              Exp (7)
                                INT: 1000000
                            PLUS
                            Exp (7)
                              ID: p3
                        RP
                      SEMI
                    StmtList (8)
                      Stmt (8)
                        Exp (8)
                          ID: hanoi
                          LP
                          Args (8)
                            Exp (8)
                              Exp (8)
                                ID: n
                              MINUS
                              Exp (8)
                                INT: 1
                            COMMA
                            Args (8)
                              Exp (8)
                                ID: p2
                              COMMA
                              Args (8)
                                Exp (8)
                                  ID: p1
                                COMMA
                                Args (8)
                                  Exp (8)
                                    ID: p3
                          RP
                        SEMI
                RC
          StmtList (10)
            Stmt (10)
              RETURN
              Exp (10)
                INT: 0
              SEMI
        RC
    ExtDefList (13)
      ExtDef (13)
        Specifier (13)
          TYPE: int
        FunDec (13)
          ID: main
          LP
          RP
        CompSt (13)
          LC
          DefList (14)
            Def (14)
              Specifier (14)
                TYPE: int
              DecList (14)
                Dec (14)
                  VarDec (14)
                    ID: sum
                  ASSIGN
                  Exp (14)
                    INT: 3
              SEMI
          StmtList (15)
            Stmt (15)
              Exp (15)
                ID: hanoi
                LP
                Args (15)
                  Exp (15)
                    ID: sum
                  COMMA
                  Args (15)
                    Exp (15)
                      INT: 1
                    COMMA
                    Args (15)
                      Exp (15)
                        INT: 2
                      COMMA
                      Args (15)
                        Exp (15)
                          INT: 3
                RP
              SEMI
            StmtList (16)
              Stmt (16)
                RETURN
                Exp (16)
                  INT: 0
                SEMI
          RC
Program
ExtDefList
ExtDef
FunDec
FUNCTION hanoi :
VarList
ParamDec
VarDec
PARAM v1
VarList
ParamDec
VarDec
PARAM v2
VarList
ParamDec
VarDec
PARAM v3
VarList
ParamDec
VarDec
PARAM v4
ComSt
DefList
StmtList
Stmt
cond_Exp
3
Exp
t1 := v1
Exp
const
INT
0
t2 := #1
IF t1 == t2 GOTO label1
GOTO label2
LABEL label1 :
Stmt
ComSt
DefList
StmtList
Stmt
*******
ID
LP
Args
RP
*******
Exp
Exp
Exp
Exp
t6 := v2
Exp
const
INT
0
t7 := #1000000
t4 := t6 * t7
Exp
t5 := v4
t3 := t4 + t5
WRITE t3
StmtList
GOTO label3
LABEL label2 :
Stmt
ComSt
DefList
StmtList
Stmt
*******
ID
LP
Args
RP
*******
Exp
Args
Exp
Exp
t10 := v1
Exp
const
INT
0
t11 := #1
t9 := t10 - t11
Args
Exp
t12 := v2
Args
Exp
t13 := v4
Args
Exp
t14 := v3
Args
